File: C#/2019-2-C샵-10주차-배열길이따기-김유빈/2019-2-C샵-10주차-배열길이따기-김유빈/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace _2019_2_C샵_10주차_배열길이따기_김유빈
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (textBox1.Text == "")
                MessageBox.Show("문장을 입력하시고 버튼을 누르세요", "경고1");
            else
            {
                int iMunjang_Giri;
                iMunjang_Giri = textBox1.TextLength;

                //textBox2.Text = iMunjang_Giri.ToString();
                string[] sMunjang_Array = new string[iMunjang_Giri];

                for (int i = 0; i <= iMunjang_Giri - 1; i++)
                {
                    sMunjang_Array[i] = textBox1.Text.Substring(i, 1); // i가 하나씩 증가한다.
                }
                textBox2.Text = "";
                for (int i = iMunjang_Giri - 1; i >= 0; i--)
                {
                    textBox2.Text = textBox2.Text + sMunjang_Array[i]; //문장을 거꾸로
                }

                //for (int i = 0; i <= iMunjang_Giri - 1; i++)
                //{
                //    textBox2.Text = sMunjang_Array[i] + textBox2.Text;
                //    //textBox2.Text = textBox2.Text + sMunjang_Array[(iMunjang_Giri - 1) -1]; //둘중 하나 사용
                //}
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            if (textBox2.Text == "")
                MessageBox.Show("문장을 입력하시고 버튼을 누르세요", "경고2");
            else
            {
                int iMunjang_Giri;
                iMunjang_Giri = textBox2.TextLength;

                //textBox2.Text = iMunjang_Giri.ToString();
                string[] sMunjang_Array = new string[iMunjang_Giri];

                for (int i = 0; i <= iMunjang_Giri - 1; i++)
                {
                    sMunjang_Array[i] = textBox2.Text.Substring(i, 1); // i가 하나씩 증가한다.
                }
                textBox3.Text = "";
                for (int i = iMunjang_Giri - 1; i >= 0; i--)
                {
                    textBox3.Text = textBox3.Text + sMunjang_Array[i]; //문장을 거꾸로=> 복호화
                }
            }
        }
    }
}

File: 비주얼 베이직/기말고사/배열-아스키/배열-아스키/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If TextBox1.Text = "" Then
            MessageBox.Show("영어문장을 소문자로 입력하세요.", "경고1")
            Exit Sub
        End If
        Dim imunjang_giri As Integer
        imunjang_giri = TextBox1.TextLength
        Dim smunjang(imunjang_giri) As String

        For j = 0 To imunjang_giri - 1
            smunjang(j) = Mid(TextBox1.Text, j + 1, 1)
            If Asc(smunjang(j)) < 97 Or Asc(smunjang(j)) > 122 Then
                MessageBox.Show("평문은 소문자로 입력하셔야 합니다.", "경고2")
                Exit Sub
            End If
        Next
        For j = 0 To imunjang_giri - 1
            If (Asc(smunjang(j)) + 3) > 122 Then
                smunjang(j) = Chr(Asc(smunjang(j)) + 3 - 122 + 96 - 32)
            Else
                smunjang(j) = Chr(Asc(smunjang(j)) + 3 - 32)
            End If
        Next
        For j = 0 To imunjang_giri - 1
            TextBox2.Text = TextBox2.Text + smunjang(j)
        Next
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        If TextBox2.Text = "" Then
            MessageBox.Show("영어문장을 대문자로 입력하세요.", "경고3")
            Exit Sub
        End If
        Dim imunjang_giri As Integer
        imunjang_giri = TextBox2.TextLength
        Dim smunjang(imunjang_giri) As String

        For j = 0 To imunjang_giri - 1
            smunjang(j) = Mid(TextBox2.Text, j + 1, 1)
            If Asc(smunjang(j)) < 65 Or Asc(smunjang(j)) > 90 Then
                MessageBox.Show("암호문은 대문자로 입력하셔야 합니다.", "경고4")
                Exit Sub
            End If
        Next
        For j = 0 To imunjang_giri - 1
            If (Asc(smunjang(j)) - 3) < 65 Then
                smunjang(j) = Chr(Asc(smunjang(j)) - 3 + 26 + 32)
            Else
                smunjang(j) = Chr(Asc(smunjang(j)) - 3 + 32)
            End If
        Next
        For j = 0 To imunjang_giri - 1
            TextBox3.Text = TextBox3.Text + smunjang(j)
        Next
    End Sub
End Class
